src/main/java/com/MyApp/budgetControl/domain/expense/category/CategoryEntity.java

```diff
@@ -0,0 +1,33 @@
+package com.MyApp.budgetControl.domain.expense.category;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Size;
+import lombok.NoArgsConstructor;
+import lombok.RequiredArgsConstructor;
+import lombok.Value;
+
+@Value
+@Entity
+@RequiredArgsConstructor
+@Table(name = "categories")
+@NoArgsConstructor(force = true)
+public class CategoryEntity {
+
+  public CategoryEntity(CategoryDTO categoryDTO) {
+    this.categoryName = categoryDTO.getCategoryName();
+  }
+
+  @Id
+  @GeneratedValue(strategy = GenerationType.AUTO)
```

Comment:
For ExpenseEntity You set id in the different way, follow the same convention everywhere

Reply:
Moved generation of category Id to the constructor level and changed it's format to a String  representation of a random UUID.